### ISSUE TITLE
command_attr is not semver compliant

### ISSUE THREAD

**Issue by SOF3** [serenity-rs/serenity#766]
## TL;DR
command_attr v0.1.7 has a compile error with serenity v0.7.4 with the following error:

```
error[E0432]: unresolved import `command_attr::group_options`
 --> /home/sofe/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.7.4/src/framework/standard/mod.rs:3:44
  |
3 |     pub use command_attr::{command, group, group_options, help, check};
  |                                            ^^^^^^^^^^^^^ no `group_options` in the root
```

## Explanation
v0.7.x [depends on command_attr `^0.1`](https://github.com/serenity-rs/serenity/blob/v0.7.x/Cargo.toml#L24-L27). When a library depending on serenity 0.7.x is fresh (without a previous Cargo.lock) dependency from crates.io, since Cargo does not respect the `path` attribute for crates.io dependencies, it tries to download the latest semver-in-range dependency, which is command_attr 0.1.7.

## Impact
Although command_attr is not expected to be used alone of serenity, users running `cargo update` or removing Cargo.lock on a project using serenity v0.7.x would encounter the aforementioned compile error.

## Suggestion
One of the following:
- Update serenity 0.7.x with a Cargo.toml that locks to `command_attr = "=0.1.6"`
- Yank command_attr v0.1.7 and release it as v0.2.0 instead. However this requires updating/yanking serenity v0.8.0 to v0.8.1 that depends on command_attr v0.2.0. Yanking versions is bad, especially when it's already 17 hours after release.

**Comment by SOF3** [comment]
By the way, it's also incorrect for serenity v0.8.0 to depend on `command_attr = "0.1"` instead of `command_attr = "0.1.7"`, since the former implies that command_attr v0.1.0 also works with serenity v0.8.0. But since nobody will downgrade command_attr for no reason, I suppose that's not a major problem.

**Comment by arqunis** [comment]
[I've taken the liberty of administering your first suggestion](https://crates.io/crates/serenity/0.7.5). This should avoid the compile errors that would've arisen after invoking `cargo update`.

I've also added [note for the future](https://github.com/serenity-rs/serenity/blob/current/Cargo.toml#L30) (your second suggestion) to address the troublesome semver incompliant version of `command_attr` for when v0.8.1 is going to be released.

Thank you for opening this issue to share awareness of this problem. It was entirely my fault; I should've paid more attention when I was publishing v0.8.0. I am sorry.
